### ISSUE TITLE
HTTP Service Editor not showing Original Response Body for echo service

### ISSUE THREAD

**Issue by avernet** [orbeon/orbeon-forms#4571]
1. Add an HTTP Service with URL `/fr/service/custom/orbeon/echo`, POSTing with XML serialization `<gaga>42</gaga>`.
2. Click Test.
3. Switch to the Test Results tab, and notice the body of the response is missing.

This works with 2019.2.2 deployed on demo.orbeon.com.

**Comment by ebruchez** [comment]
We set the serialization to `application/xml` when switching the method from `get` to `post. For this, we react to `xforms-select`. In order to know whether to set the serialization, as opposed to keeping the existing one, we check that:

1. The new value is `post` or `put`. The new value comes from `event('xxf:item-value')`.
2. The old value is `get` or `delete`. The old value comes from `.`.


Following #4438 and the associated refactoring, when the user selects the value of a single selection control, we start by computing a list of deselected and then selected values. Here we obtain:

1. `xforms-deselect` event for `get`.
2. `xforms-select` event fort `post`.

`xforms-deselect` clears the value. So when `xforms-select` is handled, `.` is blank.

We didn't use to clear the value upon `xforms-deselect`, so in effect, we introduced an incompatible change.

We introduced this change as it makes sense according to the meaning of the events, especially for `xf:copy`.
